pkgs/data/fonts/iosevka/update.sh

```diff
@@ -0,0 +1,43 @@
+#!/usr/bin/env nix-shell
+#! nix-shell -i bash -p curl gawk jq
+set -euo pipefail
+cd "$(dirname "${BASH_SOURCE[0]}")"
+
+# update_srcs $version $src_hash $bin_hash
+update_srcs() {
+    cat >srcs.nix <<EOF
+{
+  version = "$1";
+  srcSha256 = "$2";
+  binSha256 = "$3";
+}
+EOF
+}
+
+# Discover the latest version.
+nixpkgs=../../../..
+current_version=$(nix-instantiate "$nixpkgs" --eval --strict -A iosevka.version | tr -d '"')
+latest_release=$(curl --silent "https://api.github.com/repos/be5invis/Iosevka/releases/latest")
+new_version=$(jq -r '.tag_name | ltrimstr("v")' <<<"$latest_release")
+if [[ "$new_version" == "$current_version" ]]; then
+    echo "No update found"
+    exit
+else
+    echo "Updating to Iosevka $new_version"
+fi
+
+# Download and update hashes.
+# iosevka-bin needs to be built because of its usage of postFetch.
+# Related: https://github.com/NixOS/nix/issues/1514
+src_hash=$(nix-prefetch-url --unpack "https://github.com/be5invis/Iosevka/archive/v$new_version.tar.gz")
+update_srcs "$new_version" "$src_hash" "0000000000000000000000000000000000000000000000000000"
+bin_hash=$({ nix-build "$nixpkgs" --no-out-link -A iosevka-bin 2>&1 || true; } | awk -F: '/got:/ { print $3; }')
+update_srcs "$new_version" "$src_hash" "$bin_hash"
+
+# Rebuild package.json for the new version.
+package_json=$(nix-build "$nixpkgs" --no-out-link -A iosevka.src)/package.json
+jq '{name: "iosevka-build-deps", version, dependencies}' "$package_json" >package.json
```

Comment:
I think what I get from here is that it's not available on npm? That's why you can't put it in `node-packages.json`? If so, you should know it's possible to do this:

https://github.com/NixOS/nixpkgs/blob/a38590d91ce79e56d1bae79c5824008e906813ef/pkgs/development/node-packages/node-packages.json#L145

And you then will need only to update the version tag there every once in a while.. If that will be possible, you should be able to remove a lot of moving parts from these derivations.

Reply:
Well, that does indeed simplify things a lot. I had to use a tarball though, as node2nix doesn't perform a shallow fetch and Iosevka's repo is 1GB+ in size.